presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -460,6 +472,50 @@ private void checkTypesMatchForInsert(Insert insert, Scope queryScope, List<Colu
             }
         }
 
+        private void checkTypesMatchForNestedStructs(
+                Node node,
+                String errorMessage,
+                int i,
```

Comment:
nit: How about naming this parameter `columnIndex`? `i` is very vague if we look at this function alone.

Reply:
made the change